regcomp.c

```diff
@@ -8155,7 +8155,8 @@ S_handle_possible_posix(pTHX_ RExC_state_t *pRExC_state,
                 p++;
             }
             else {
-                input_text[name_len++] = utf8_to_uvchr_buf((U8 *) p, e, NULL);
+                input_text[name_len++] = utf8_to_uv_or_die((U8 *) p, (U8 *) e,
```

Comment:
While it was in the original, there's no need to cast away const here, `p`  and `e` are `const char *`, `s` and `e` for utf8_to_uv_or_die() are `const U8 *`.